### PR TITLE
Add support for JetBrains Rider on Windows

### LINKED ISSUE #9365: Add support for JetBrains Rider
Currently, JetBrains Rider is not supported as an external editor.

### PR DESCRIPTION
**Pull request by say25**
Closes #9365

## Description

- Add support for JetBrains Rider on Windows

## Release notes

Notes: Add support for JetBrains Rider on Windows

### CODE REVIEW

File: app/src/lib/editors/win32.ts [hunk 1]
```diff
@@ -645,6 +664,37 @@ function extractApplicationInformation(
...
+  }
+
+  if (editor === ExternalEditor.Rider) {
+    let displayName = ''
+    let publisher = ''
+    let installLocation = ''
+
+    for (const item of keys) {
+      // NOTE:
+      // Webstorm adds the current release number to the end of the Display Name, below checks for "PhpStorm"
```
Comment (say25) [inline app/src/lib/editors/win32.ts]:
```suggestion
      // Webstorm adds the current release number to the end of the Display Name, below checks for "JetBrains Rider"
```

**Comment by tierninho** [comment]
@say25 Thanks for implementing the Windows side. I spotted this in the logs when trying to activate a repo/file from Desktop, if a repo/file was already active in Rider in the background. 

Repro:
- Open a file from Repo X in Rider
- Open the same (entire) repo in Repo X
- Desktop remains in the foreground, Rider does not open repo, and error surfaced in logs

```
[11396:0330/090924.127:ERROR:CONSOLE(109)] "Uncaught (in promise) Error: Could not instantiate: ProductRegistryImpl.Regi
stry", source: devtools://devtools/bundled/shell.js (109)
```

**Comment by van800** [comment]
Might be worth using https://github.com/JetBrains/resharper-unity/blob/net201/unity/EditorPlugin/RiderPathLocator.cs

**Comment by say25** [comment]
@tierninho thats kinda odd, are you using `Rider 2019.3.4`+ it worked for me.

image

I then opened the project, opened and opened files commits from commits in the history tab.

**Comment by say25** [comment]
@tierninho I think Rider might play nicer with .Net Project. Feel free to use this one as a test https://github.com/PostgreSQLCopyHelper/PostgreSQLCopyHelper

**Pull request merged**